src/App.tsx

```diff
@@ -15,49 +19,97 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Name = 'name',
+  Length = 'length',
+}
+
+function getPreparedGoods(
+  goods:string[],
+  { sortField, confirmation }: FilterParams,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.Name:
+          return good1.localeCompare(good2);
+        case SortType.Length:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (confirmation) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<string>('');
+  const [confirmation, setConfirmation] = useState(false);
+  const visibleGoods
+    = getPreparedGoods(goodsFromServer, { sortField, confirmation });
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={cn('button', 'is-info', {
+            'is-ligth': sortField !== SortType.Name,
+          })}
+          onClick={() => {
+            setSortField(SortType.Name);
+          }}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={cn('button', 'is-success', {
+            'is-ligth': sortField !== SortType.Length,
+          })}
+          onClick={() => {
+            setSortField(SortType.Length);
+          }}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={cn('button', 'is-warning', {
+            'is-ligth': !confirmation,
+          })}
+          onClick={() => {
+            setConfirmation(!confirmation);
+          }}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(sortField || confirmation)
+          && (
+            <button
+              type="button"
+              className="button is-danger is-light"
+              onClick={() => {
+                setSortField('');
+                setConfirmation(false);
+              }}
```

Comment:
Move this logic from jsx to the helper function and use it here
```suggestion
              onClick={reset}
```